#! /usr/bin/env python


# Initialize ros and import parameter generator
PACKAGE='motion_force_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level Description     	Default 	Min   	Max
gen.add("convergenceRate",double_t,0,"Convergence scale",5.0,1.0,10.0)
gen.add("filteredForceGain",double_t,0,"filtering weight of force torque sensor",0.9,0,1)
gen.add("contactForceThreshold",double_t,0,"Force threshold used to detect contact [N]",6,0,10)
gen.add("controlForce",bool_t,0,"Activate force control",False)
gen.add("targetForce",double_t,0,"Target force to apply [N]",5.0,0.0,20.0)
gen.add("polishing",bool_t,0,"Activate polishing motion",False)
gen.add("linear",bool_t,0,"Activate linear motion",False)
gen.add("k1",double_t,0,"k1",10,0.0,50.0)
gen.add("k2",double_t,0,"k2",1,0.0,10.0)
gen.add("minFc",double_t,0,"min Fc",-15.0,-50.0,0.0)
gen.add("maxFc",double_t,0,"max Fc",30.0,0.0,50.0)
gen.add("lambda1",double_t,0,"lambda 1",0.0,0.0,1000.0)
gen.add("lambda2",double_t,0,"lambda 2",0.0,0.0,1000.0)

exit(gen.generate(PACKAGE, "motion_force_control", "unifiedMotionForceController_params"))